source/extensions/dynamic_modules/sdk/rust/src/lib.rs

```diff
@@ -363,6 +363,114 @@ pub trait EnvoyHttpFilter {
   ///
   /// Returns true if the operation is successful.
   fn set_dynamic_metadata_string(&mut self, namespace: &str, key: &str, value: &str) -> bool;
+
+  /// Get the currently buffered request body. The body is represented as a list of [`EnvoyBuffer`].
+  /// Memory contents pointed by each [`EnvoyBuffer`] is mutable and can be modified in place.
```

Comment:
In looking at this more it seems kind of confusing that EnvoyBuffer supports both mut and not mut access, as I don't think mutability is safe or works correctly in all cases? For example header map stuff? Is it safe in all cases even if there is a mutable borrow on the filter? Do we need to split the EnvoyBuffer into a read only and mutable type?

/wait-any

Reply:
>  Do we need to split the EnvoyBuffer into a read only and mutable type?

yeah agree this is confusing and I was thinking about the exact same thing about having two EnvoyBuffers kinds for mutable and non-mutable version. I think the headers should be returned with immutable version. I will follow up in the separate PR in a bit

Reply:
OK sounds good you can fix in a follow up. Personally I I would call this one EnvoyMutBuffer and leave the other one as is.